web/session.go

```diff
@@ -125,6 +125,35 @@ func loginOK(r *rest.Request) bool {
 	return true
 }
 
+func loginWithTokenOK(r *rest.Request, token string) bool {
+	restToken, err := auth.ParseRestToken(token)
+	//validToken, err := auth.ValidateRestToken(r.Request, token)
+	if err != nil {
+		msg := "Unable to parse rest token"
+		plog.WithError(err).Info(msg)
+		return false
+	} else if !restToken.HasAdminAccess() {
+		msg := "Could not login with rest token. Insufficient permissions."
+		plog.Info(msg)
```

Comment:
Actually definitely debug all of these, and use `plog.WithField("url", r.url)`. `r.url` may not be a thing, but however you get the url from the request.